src/Runner.Worker/ActionManager.cs

```diff
@@ -355,6 +396,26 @@ public Definition LoadAction(IExecutionContext executionContext, Pipelines.Actio
                         Trace.Verbose($"Details: {StringUtil.ConvertToJson(compositeAction?.Steps)}");
                         Trace.Info($"Load: {compositeAction.Outputs?.Count ?? 0} number of outputs");
                         Trace.Info($"Details: {StringUtil.ConvertToJson(compositeAction?.Outputs)}");
+                        if (CachedChildPreSteps.TryGetValue(action.Id, out var preSteps))
+                        {
+                            compositeAction.PreSteps = preSteps;
+                        }
+                        if (CachedChildPostSteps.TryGetValue(action.Id, out var postSteps))
+                        {
+                            compositeAction.PostSteps = postSteps;
+                        }
+                        if (_cachedChildStepIds.ContainsKey(action.Id))
```

Comment:
consider "embedded" terminology for consistency with other areas (e.g. execution context)

Reply:
consider computing the IDs (stable IDs)... would be consistent with the server. See IdUtility.cs and InstanceRef.cs on the server

Reply:
Could add "GetEmbeddedStepInstanceId"